### PR TITLE
Bug 4198/v3

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/4198

suricata-verify-pr: 448

### CODE REVIEW

File: rust/src/dcerpc/dcerpc.rs [hunk 1]
```diff
@@ -1197,7 +1205,15 @@ pub extern "C" fn rs_dcerpc_parse_response(
     if input_len > 0 {
         if input != std::ptr::null_mut() {
             let buf = build_slice!(input, input_len as usize);
-            return state.handle_input_data(buf, core::STREAM_TOCLIENT);
+            let ret = state.handle_input_data(buf, core::STREAM_TOCLIENT);
+            let tx_id = state.tx_id - 1;
```
Comment (victorjulien) [inline rust/src/dcerpc/dcerpc.rs:1209]: Is this correct? I would expect a tx to be created on the request side, not the response. So here I would expect the `state.tx_id` to stay the same.

File: rust/src/dcerpc/dcerpc.rs [hunk 2]
```diff
@@ -1177,14 +1177,22 @@ pub extern "C" fn rs_dcerpc_parse_request(
     }
     if input_len > 0 && input != std::ptr::null_mut() {
         let buf = build_slice!(input, input_len as usize);
-        return state.handle_input_data(buf, core::STREAM_TOSERVER);
+        let ret = state.handle_input_data(buf, core::STREAM_TOSERVER);
+        let tx_id = state.tx_id - 1;
+        let req_done = if let Some(tx) = state.get_tx(tx_id) {
```
Comment (victorjulien) [inline rust/src/dcerpc/dcerpc.rs:1182]: for the record: if this works it needs to be cleaned up to avoid the tx lookup

Comment (inashivb) [reply rust/src/dcerpc/dcerpc.rs:1182]: OK. On it.

**Comment by inashivb** [comment]
Replaced with #5889

**Pull request closed without merging**